rust/derive/src/utils.rs

```diff
@@ -0,0 +1,75 @@
+/* Copyright (C) 2021 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/// If we're being used from within Suricata we have to reference the internal name space with
+/// "crate", but if we're being used by a library or plugin user we need to reference the
+/// Suricata name space as "suricata". Check the CARGO_PKG_NAME environment variable to
+/// determine what identifier to setup.
+pub fn crate_id() -> syn::Ident {
+    let is_suricata = std::env::var("CARGO_PKG_NAME")
+        .map(|var| var == "suricata")
+        .unwrap_or(false);
+    if is_suricata {
+        syn::Ident::new("crate", proc_macro2::Span::call_site())
+    } else {
+        syn::Ident::new("suricata", proc_macro2::Span::call_site())
+    }
+}
+
+/// Transform names such as "OneTwoThree" to "one_two_three".
+pub fn transform_name(name: &str, delim: char) -> String {
```

Comment:
The move and the change of behaviour exist in one commit where it should probably be 2 and document why the change is needed in the commit message.